src/main/java/org/jclouds/vcloud/director/v1_5/compute/functions/ImageForVAppTemplate.java

```diff
@@ -86,6 +90,17 @@ public Image apply(QueryResultVAppTemplateRecord from) {
       }
       builder.operatingSystem(os);
       builder.status(toPortableImageStatus.apply(from.getStatus()));
+
+      Reference reference = Reference.builder().name(from.getVdc()).href(URI.create(from.getVdc())).build();
+
+      if (reference != null) {
+         try {
+            builder.location(findLocationForResource.apply(reference));
+         } catch (NoSuchElementException e) {
+            logger.debug(e.getMessage());
```

Comment:
maybe it's a warning. add a message to warn users about the issue. something like, logger.warn("the image id doesn't have a location metadata attached and may lead to priority issues", e);